sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubServiceClient_helpers.cs

```diff
@@ -354,5 +382,63 @@ private string GenerateTokenFromAzureKeyCredential(DateTimeOffset expiresAt, str
 
             return jwt.BuildString();
         }
+
+        private string GetRelativeClientEndpoint(WebPubSubClientType endpointType) => endpointType switch
+        {
+            WebPubSubClientType.Default => $"client/hubs/{_hub}",
+            WebPubSubClientType.Mqtt => $"clients/mqtt/hubs/{_hub}",
+            _ => throw new ArgumentOutOfRangeException(nameof(endpointType))
+        };
+
+        private async Task<string> GetClientAccessTokenCore(
+             DateTimeOffset expiresAt,
+             string userId,
+             IEnumerable<string> roles,
+             IEnumerable<string> groups,
+             WebPubSubClientType clientType,
+             bool async,
+             CancellationToken cancellationToken = default)
+        {
+            if (clientType == WebPubSubClientType.Mqtt && _apiVersionEnum < WebPubSubServiceClientOptions.ServiceVersion.V2024_01_01)
+            {
+                throw new NotSupportedException($"Getting client access token for MQTT is supported since API version {WebPubSubServiceClientOptions.ServiceVersion.V2024_01_01.ToVersionString()}. Current version is {_apiVersion}.");
```

Comment:
Nit - you probably want to use `_apiVersion.ToString()` so both values are in the same format.  I'd also call it a `client access URI` rather than `token` since we only expose the URI part in our public API surface.  Maybe we could make it look something like?
```suggestion
                throw new NotSupportedException($"Generating a client access URI for MQTT is only supported in API version {WebPubSubServiceClientOptions.ServiceVersion.V2024_01_01.ToVersionString()} or later.  You are currently using API version {_apiVersion.ToString()}.");
```

Reply:
Updated